alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt

```diff
@@ -115,22 +115,29 @@ class TransportGetAlertsAction @Inject constructor(
 
         if (getAlertsRequest.monitorId != null) {
             queryBuilder.filter(QueryBuilders.termQuery("monitor_id", getAlertsRequest.monitorId))
-            if (getAlertsRequest.workflowIds.isNullOrEmpty()) {
+            if (
+                getAlertsRequest.workflowIds != null && getAlertsRequest.workflowIds!!.size == 1 && getAlertsRequest.workflowIds!![0] == ""
```

Comment:
Nitpick: Looks like this check is duplicated in a few spots. It could be helpful to extract it into a helper method that has a doc comment explaining it further.

Reply:
extracted into method and added comment